docker/docker-compose.testing.yml

```diff
@@ -0,0 +1,28 @@
+version: '3'
+services:
+  db:
+    image: postgres:12
+    volumes:
+      - './db:/var/lib/postgresql/data'
+    ports:
+      - '5432:5432'
+    environment:
+      POSTGRES_USER: farm
+      POSTGRES_PASSWORD: farm
+      POSTGRES_DB: farm
+
+  www:
+    depends_on:
+      - db
+    image: farmos/farmos:2.x-dev
+    volumes:
+      - './www:/opt/drupal'
+
+  test-runner:
+    depends_on:
+      - www
+    image: farmos/farmos:2.x
+    entrypoint: /bin/sh
+    command: -c 'sleep 10 && exec docker-entrypoint.sh apache2-foreground'
+    volumes:
+      - './www:/opt/drupal'
```

Comment:
Having a dedicated "testing" docker compose file will provide a place to start up additional database types to be tested or other testing-specific customizations in the future - which may deviate from the recommended "development" containers.

Reply:
Great! I like this approach a lot.

Question: what is the `sleep 10` for? And how does it relate to the `sleep 15` in the workflow itself? Do we need this one too?

Is it related to the `depends_on: www` statement?

Are they both there to ensure the `www` has a chance to populate the the codebase with a dev version before the `test-runner` container does? That makes sense, but I wonder if there's a small chance that it is inconsistent. We can cross that bridge if/when we come to it... not necessary to change anything now, I don't think.

Reply:
> Are they both there to ensure the www has a chance to populate the the codebase with a dev version before the test-runner container does?

Yes. I toyed with some sort of loop to wait until the directory exists, but that didn't seem very conclusive since we actually want to wait until the full copy has occurred. That could be signalled via a sentinel file, however I didn't want to get into that in this change since it would add a fair bit of complexity.

Reply:
> Question: what is the `sleep 10` for? And how does it relate to the `sleep 15` in the workflow itself? Do we need this one too?

The two sleeps occur in parallel. The relationship between them is that the one in the workflow must be larger than the one here.